torchdata/stateful_dataloader/stateful_dataloader.py

```diff
@@ -909,16 +930,17 @@ def __init__(self, loader, next_iter_state):
                 next(self)
             self._finished = next_iter_state[_ITERATOR_FINISHED]
 
-    def _reset_state_vars(self):
-        self._worker_snapshots = {}
+    def _reset_state_vars(self, worker_states):
+        assert len(worker_states) == self._num_workers, (self._num_workers, worker_states)
+        self._worker_snapshots = {key: _IncrementalWorkerState(state) for key, state in worker_states.items()}
```

Comment:
This is where the worker snapshots are initialized now, worker_states will either contain t0 worker state or whatever is being loaded